### DISCUSSION TITLE
Custom SCSS not built with the command "build-keycloak-theme"

### DISCUSSION THREAD

**Discussion by Ayuro** [keycloakify/keycloakify#703] [Q&A]
Hello there,

I'm working on a project to custom a Keycloak login page so it look like an old version of the client' site login page.
So I prepare the env and made small changes before trying to upload and apply the new theme to our Keycloak test env.
But my changes seems to not being applied.

I've made a main.scss and put it in the login folder. I import it in the KcPage.tsx file.

When I run the storybook, I can see the changes.
When I use build-keycloak-theme, into dist/assets I have a file named KcPage-"random letters".css with my changes and more. But when I look into the .jar file stocked in dist_keycloak, the .css file (named login.css) does not contain my changes.

I'm probably doing something wrong, but what could it be?
Does anyone could help please?

If you need more informations please, do not hesitate :)

**Comment by garronej** [comment]
Hello @Ayuro,  

Please provide a simple modification of the vite starter repo with the described changes that doesn't apply.  
If the setup aint right I'll tell you what's up. If there's a bug in Keycloakify I'll fix it but this is very unlikely. Such an important bug would have been reported before.

**Reply by Ayuro** [reply]
Hope this is what you needed ^^
I've made a public repo: https://github.com/Ayuro/kcify-starter.git

The edited file is the login.stories.tsx
The scss can be found in src/login/main.scss

I did not touched any other files.

If you needed something more specific tell me.

**Reply by garronej** [reply]
Thank you!  

Everything is working as expected. The theme renders in Keycloak the same way it does in Storybook:  

<img width="2056" alt="image" src="https://github.com/user-attachments/assets/b6e31ffb-b20e-4c7f-b3b4-967ea9912828">

You can try running `npx keycloakify start-keycloak` to verify.  

It seems the issue may be related to how you're loading your theme into Keycloak. You can find documentation on this process here:  
https://docs.keycloakify.dev/importing-your-theme-in-keycloak  

PS: After reviewing your issue again, it seems like the problem might actually be figuring out how the JAR works. If that is the case, well it's complicated. There is a lot of work performed by Keycloakify during the build process to make sure everyting works as expected. Reverse engeneering how it works just by looking at the content of the JAR isn't trivial.

**Reply by Ayuro** [reply]
We'll try again but yesterday evening we've try to set the new theme on our test keycloak page to be sure it was working before I made some heavy changes and it kept the main keycloak theme. Only the favicon was changeing.

I'll check that all out and I keep you posted if anything change.

PS: No, we are not trying to figure out how the JAR works. It was just an indication on were we lost trace of our CSS, but that's probably normal.
Our problem is more, in storybook we see our changes, but when we follow the steps to import and implement our theme (via JAR) to our testing keycloak and we select the them, only the favicon change, we do not have the change of shape and color as you have in your screenshot.

Thanks for the quick response tho.

**Reply by garronej** [reply]
First thing to try is to run `npx keycloakify start-keycloak` and navigate to `https://my-theme.keycloakify.dev` If it renders corectly there. If it does it's just a matter of making sure to properly import and enable the theme in your production keycloak:  

https://docs.keycloakify.dev/enabling-your-theme  
https://docs.keycloakify.dev/importing-your-theme-in-keycloak

**Reply by Ayuro** [reply]
It's work now.
But for information, that's what we were doing:

We were running the prompt "npm run build-keycloak-theme".
It generate our "dist_keycloak" folder with the file keycloak-theme-for-kc-22-to-25.jar which we take and put into the "provider" folder in our keycloak instance.

Then, on our dashboard the theme "keycloakify-starter" appeared so we select it.
It was charged because the favicons was the one from Keycloakify but the changes we made was not.

We try now by injecting direclty the "keycloakify-starter" folder (that is generated by the command "npx keycloakify start-keycloak") found in "dist_keycloak/theme" into the "theme" folder on our keycloak instance and it work.

We thought the "good practice" was to use the .JAR but if it work the other way it's ok.

Thanks you for the help.

**Reply by garronej** [reply]
> We try now by injecting direclty the "keycloakify-starter" folder (that is generated by the command "npx keycloakify start-keycloak") found in "dist_keycloak/theme" into the "theme" folder on our keycloak instance and it work.  

No no no! Don't do that. You'll run into issues. You can't just copy the the directory, some things won't work like the password policies validation.  

> We thought the "good practice" was to use the .JAR but if it work the other way it's ok.

This **is** the correct approach. Maybe there was just cache issue causing the jar to be corrupted for some reason I don't know. Just delete your node_modules and `keycloak_dist` just in case and build again.

**Reply by Ayuro** [reply]
Oh, ok, we'll do that then. 
Thanks you
